lib/jsdom/living/attributes.js

```diff
@@ -30,6 +31,7 @@ exports.changeAttribute = (element, attribute, value) => {
   const { _localName, _namespace, _value } = attribute;
 
   queueAttributeMutationRecord(element, _localName, _namespace, _value);
+  invalidateStyleCache(element);
```

Comment:
I think all, or maybe almost-all, `invalidStyleCache` calls can be moved into https://github.com/jsdom/jsdom/blob/22f7c3c51829a6f14387f7a99e5cdf087f72e685/lib/jsdom/living/nodes/Node-impl.js#L215